backend/inspirehep/disambiguation/tasks.py

```diff
@@ -79,57 +79,35 @@ def match_literature_author_with_config(author_data, matcher_config):
     matched_records = [
         matched_record
         for matched_record in match(author_data, matcher_config)
-        if matched_record.get("inner_hits")
+        if get_value(
+            matched_record, "inner_hits.authors.hits.hits[0]._source.record.$ref"
+        )
     ]
     return matched_records
 
 
-def get_record_refs_for_matched_literature_authors(matched_records):
-    matched_authors_refs = set()
-    for record_authors in matched_records:
-        matched_authors_refs.update(
-            get_value(
-                record_authors, "inner_hits.authors.hits.hits._source.record.$ref", None
-            )
-        )
-    return matched_authors_refs
-
-
-def assign_reference_to_author_if_unambiguous_match(matched_authors_refs, author):
-    if matched_authors_refs and len(matched_authors_refs) == 1:
-        author_reference_url = matched_authors_refs.pop()
-        author["record"] = {"$ref": author_reference_url}
-        return author_reference_url.split("/")[-1]
-
-
-def assign_reference_to_author_if_unambiguous_literature_author_match(
-    matched_records, author
-):
-    matched_record_list = list(matched_records)
-    matched_authors_refs = get_record_refs_for_matched_literature_authors(
-        matched_record_list
-    )
-    matched_ref = assign_reference_to_author_if_unambiguous_match(
-        matched_authors_refs, author
-    )
-    if matched_ref:
-        author_ids = get_value(
-            matched_record_list[0], "inner_hits.authors.hits.hits._source.ids", []
-        )
-        assign_bai_to_author(author_ids, author)
-        return matched_ref
+def get_reference_and_bai_if_unambiguous_match(matched_refs_ids):
+    if len(matched_refs_ids.keys()) == 1:
```

Comment:
The length of a `dict` is defined by the number of keys:
```suggestion
    if len(matched_refs_ids) == 1:
```